### DISCUSSION TITLE
Custom maven image repository

### DISCUSSION THREAD

**Discussion by mxlt02** [PaperMC/Paper#10763] [Ideas]
### Is your feature request related to a problem?

minecraft is a world famous game, paper core is also very famous, However, the default image repository"repo.maven.apache.org" used by the built-in SpigotLibraryLoader is too slow to be used in many places.

### Describe the solution you'd like.

Custom maven image repository

### Describe alternatives you've considered.

Provides a configuration item

### Other

Thank you all for your hard work

**Comment by notTamion** [comment]
you can load dependencies from custom repositories using Paper Plugins

**Reply by mxlt02** [reply]
> you can load dependencies from custom repositories using Paper Plugins

Do I need to develop it myself? I haven't found similar plugins

**Reply by masmc05** [reply]
the plugin author needs to add this support, not you

Ask them to add the support for custom repository

**Reply by mxlt02** [reply]
> the plugin author needs to add this support, not you
> 
> Ask them to add the support for custom repository

Thank you. This might be a little hard

**Comment by electronicboy** [comment]
> used by the built-in SpigotLibraryLoader is too late to be used in many places

I'm not sure what is even meant by this. I do and have supported the notion of letting people add a custom repo to serve as a proxy to central, be that to have a reliable "local" cache of central or for people running private servers to be able to deploy libraries to, but, outside of those two concerns, there is no real big thing here for allowing a custom repo. It's not a saving grace to plugin devs because they'd need their end users to keep switching repos every 2 weeks as they upgrade plugins to pull in their deps, etc.

for plugins themselves, paper plugins, or looking into library injection tools on their own, are generally going to be the only "wider" solution here

**Reply by mxlt02** [reply]
Sorry I didn't make it clear, I meant that the official image download speed is slow in some regions of the world, thank you for your reply
